core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -20921,17 +20966,9 @@ LogicalProject(DEPTNO=[$0])
     <Resource name="planAfter">
       <![CDATA[
 LogicalProject(SAL=[$5])
-  LogicalFilter(condition=[OR(=($9, 0), IS NOT TRUE(OR(IS NOT NULL($13), <($10, $9))))])
-    LogicalJoin(condition=[AND(=($0, $12), =($2, $14))], joinType=[left])
-      LogicalJoin(condition=[=($2, $11)], joinType=[left])
-        LogicalTableScan(table=[[CATALOG, SALES, EMP]])
-        LogicalProject(c=[CASE(IS NOT NULL($2), $2, 0)], ck=[CASE(IS NOT NULL($2), $2, 0)], NAME=[$0])
-          LogicalJoin(condition=[IS NOT DISTINCT FROM($0, $1)], joinType=[left])
-            LogicalAggregate(group=[{2}])
-              LogicalTableScan(table=[[CATALOG, SALES, EMP]])
-            LogicalAggregate(group=[{0}], c=[COUNT()])
-              LogicalProject(NAME=[$1])
-                LogicalTableScan(table=[[CATALOG, SALES, DEPT]])
+  LogicalFilter(condition=[IS NULL($10)])
```

Comment:
these plans are significantly simpler, I hope they are correct too.. not easy to tell

Reply:
I have tried to explain step by step how I verified that the rewrite is sound (not a formal proof but it will help increase confidence in the change): https://github.com/apache/calcite/pull/4668#issuecomment-3618906931